cli/templatepull.go

```diff
@@ -36,39 +39,67 @@ func (r *RootCmd) templatePull() *clibase.Cmd {
 				dest = inv.Args[1]
 			}
 
-			// TODO(JonA): Do we need to add a flag for organization?
 			organization, err := CurrentOrganization(inv, client)
 			if err != nil {
-				return xerrors.Errorf("current organization: %w", err)
+				return xerrors.Errorf("get current organization: %w", err)
 			}
 
 			template, err := client.TemplateByName(ctx, organization.ID, templateName)
 			if err != nil {
-				return xerrors.Errorf("template by name: %w", err)
+				return xerrors.Errorf("get template by name: %w", err)
 			}
 
-			// Pull the versions for the template. We'll find the latest
-			// one and download the source.
-			versions, err := client.TemplateVersionsByTemplate(ctx, codersdk.TemplateVersionsByTemplateRequest{
-				TemplateID: template.ID,
-			})
-			if err != nil {
-				return xerrors.Errorf("template versions by template: %w", err)
-			}
+			var latestVersion codersdk.TemplateVersion
+			{
+				// Determine the latest template version and compare with the
+				// active version. If they aren't the same, warn the user.
+				versions, err := client.TemplateVersionsByTemplate(ctx, codersdk.TemplateVersionsByTemplateRequest{
+					TemplateID: template.ID,
+				})
+				if err != nil {
+					return xerrors.Errorf("template versions by template: %w", err)
+				}
+
+				if len(versions) == 0 {
+					return xerrors.Errorf("no template versions for template %q", templateName)
+				}
+
+				// Sort the slice from newest to oldest template.
+				sort.SliceStable(versions, func(i, j int) bool {
+					return versions[i].CreatedAt.After(versions[j].CreatedAt)
+				})
 
-			if len(versions) == 0 {
-				return xerrors.Errorf("no template versions for template %q", templateName)
+				latestVersion = versions[0]
 			}
```

Comment:
You should move this into the switch when you pick latest

Reply:
I need to know the latest version for the warning when the version isn't specified.